### ISSUE TITLE
Current search feature performs "LIKE" comparison operation on ALL searchable tagged fields

### ISSUE THREAD

**Issue by jmachen** [ActiveChooN/gin-gorm-filter#10]
As this seems to be functioning as intended I would suggest adding a separate SEARCH_SINGLE feature that performs the LIKE operation against only the specified field with supporting code that looks like the following (which just applies the same column specific filtering used by `filterField`):

```golang
func searchSingleField(field reflect.StructField, phrase string) clause.Expression {
	var paramName string
	if !strings.Contains(field.Tag.Get(tagKey), "searchable") {
		return nil
	}

	columnName := getColumnNameForField(field)
	paramMatch := paramNameRegexp.FindStringSubmatch(field.Tag.Get(tagKey))
	if len(paramMatch) == 2 {
		paramName = paramMatch[1]
	} else {
		paramName = columnName
	}
	re, err := regexp.Compile(fmt.Sprintf(`(?m)%v:(\w{1,}).*`, paramName))
	if err != nil {
		return nil
	}
	searchSubPhraseMatch := re.FindStringSubmatch(phrase)
	if len(searchSubPhraseMatch) == 2 {
		return clause.Like{Column: columnName, Value: "%" + searchSubPhraseMatch[1] + "%"}
	}

	return nil
}
```

**Comment by ActiveChooN** [comment]
@jmachen, I think it can be done by extending filter parameter with `~` operator that can be used for `SEARCH` query. wdyt?

**Comment by jmachen** [comment]
So something like:
`https://myurl.com/api/v1/widgets?search=field~value`

I think that's perfectly reasonable and keeps the URL length from growing unnecessarily.

**Comment by jmachen** [comment]
I'll submit a PR within the next day or so.

**Comment by ActiveChooN** [comment]
Closed with #13
